### ISSUE TITLE
Add-to-app manual embedding (option C in the documentation) builds FlutterPluginRegistrant.xcframework as a static library

### ISSUE THREAD

**Issue by ayansg** [flutter/flutter#94700]
As described in the [documentation of add-to-app scenario (Option C)](https://docs.flutter.dev/development/add-to-app/ios/project-setup#option-c---embed-application-and-plugin-frameworks-in-xcode-and-flutter-framework-with-cocoapods) when we need to manually embed a Flutter module to an iOS host app, we are supposed to **embed** "FlutterPluginRegistrant.xcframework" to the host app bundle. 

<img width="958" alt="Screenshot 2021-12-06 at 5 13 33 AM" src="https://user-images.githubusercontent.com/1812245/144768885-bdc852ea-ff06-4aff-b1dc-1f8f8e556862.png">

However, in my own experience this specific framework is actually created as a **static** framework when we run `flutter build ios-framework --cocoapods --output=/path/to/output/directory` with the latest "stable" release of Flutter (v2.5.3) and should not be embedded as such. I realized this fun fact when I tried to ship an iOS app containing a flutter module, added through Option C of documentation. Xcode reported the following error:
> Found an unexpected Mach-O header code: 0x72613c21

and this essentially means that I had embedded a ***static*** framework somewhere in my app bundle. So I ran `file` command to all framework binaries and finally realized that the above one was the culprit. And I believe this will affect Option B of add-to-app module integration scenarion as well (haven't tried that though).

<img width="1117" alt="Screenshot 2021-12-06 at 5 25 47 AM" src="https://user-images.githubusercontent.com/1812245/144769331-6fbb0624-94db-4041-a288-6a8f90d7c38a.png">

Consequently, I updated the framework linkage from "link + embed" to "link only" and everything worked as expected. So it will be helpful to know if this an intended change in the build tools for iOS that is not documented yet or it is just another known issue or it is me, doing something really weird causing this. I'll appreciate any feedback on this.

Output of `flutter doctor -v`
===
asengupta@Ayans-MacBook-Pro mica % flutter doctor -v
[✓] Flutter (Channel stable, 2.5.3, on macOS 12.0.1 21A559 darwin-x64, locale en-GB)
    • Flutter version 2.5.3 at /Users/asengupta/Development/FlutterSDK/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (7 weeks ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/asengupta/Library/Android/sdk
    • Platform android-31, build-tools 30.0.2
    • ANDROID_HOME = /Users/asengupta/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.1, Build version 13A1030d
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.62.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (5 available)
    • Redmi 4 (mobile)       • 6b90d397d440                             • android-arm64  • Android 7.1.2 (API 25)
    • Ayan’s iPhone (mobile) • 4c5147f0659468f98e0102e8a0d2673d5d7ca141 • ios            • iOS 15.1 19B74
    • iPhone 8 (mobile)      • C52B73E0-8DD3-4DE3-98B5-8297D3077B5A     • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-0 (simulator)
    • macOS (desktop)        • macos                                    • darwin-x64     • macOS 12.0.1 21A559 darwin-x64
    • Chrome (web)           • chrome                                   • web-javascript • Google Chrome 96.0.4664.55
    ! Error: Ayan’s iPhone is busy: Fetching debug symbols for Ayan’s iPhone. Xcode will continue when Ayan’s iPhone is finished. (code -10)
    ! Error: Ayan’s NGS iPhone is not connected. Xcode will continue when Ayan’s NGS iPhone is connected. (code -13)

• No issues found!

Xcode version
===
Xcode 13.1

CocoaPods version
===
1.11.2

**Comment by maheshmnj** [comment]
Leaving this issue open for further insights from the team.

cc: @jmagman

**Comment by jmagman** [comment]
You're right, this a documentation issue.  [Option B](https://docs.flutter.dev/development/add-to-app/ios/project-setup#embed-the-frameworks) mentions this:

> For example, you can drag the framework (except for FlutterPluginRegistrant and any other static frameworks)

and

> Important: Plugins might produce static or dynamic frameworks. Static frameworks should be linked on, but never embedded. If you embed a static framework into your application, your application is not publishable to the App Store and fails with a Found an unexpected Mach-O header code archive error.
